teps/0121-refine-retries-for-taskruns-and-customruns.md

```diff
@@ -129,14 +129,34 @@ Task-level Timeout (`TaskRunSpec.Timeout` and `RunSpec.Timeout`) is set for each
 
 ### Retries in TaskRuns and CustomRuns
 
-Add a new `Retries` field to [TaskRunSpec](https://pkg.go.dev/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1#TaskRunSpec). Model the `CustomRunSpec` based on the `RunSpec` to use the existing `retries` field.
+Add a new `Retries` field in [TaskRunSpec](https://pkg.go.dev/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1#TaskRunSpec). Model the `CustomRunSpec` based on the `RunSpec` to use the existing `retries` field.
 
 The `PipelineTask.Retries` value, which is specified at `Pipeline` authoring time, is passed to the `TaskRunSpec.Retries` and `CustomRunSpec.Retries` during the execution of a `PipelineRun`.
 
 The `TaskRun` and `CustomRun` controllers handle their own `Retries`. 
 
 The `PipelineRun` controller does not check for `len(retriesStatus)` to determine whether a `TaskRun` or `CustomRun` is done executing. Instead, uses `ConditionSucceeded` as the only way to decide if the `TaskRun` or `CustomRun` has completed execution.
 
+#### Execution Status of a `failed` `pipelineTask` with `retries`
+
+The following table shows how an overall status of a `taskRun` or a `customRun` for a `pipelineTask` with `retries` set to 3:
+
+| `status` | `reason`    | description                                                                                                |
+|----------|-------------|------------------------------------------------------------------------------------------------------------|
+| Unknown  | Running     | The `taskRun` has been validated and started to perform its work.                                          |
+| Unknown  | ToBeRetried | The `taskRun` (zero attempt of a `pipelineTask`) finished executing but failed. It has 3 retries pending.  |
+| Unknown  | Running     | First attempt of a `taskRun` has started to perform its work.                                              |
+| Unknown  | ToBeRetried | The `taskRun` (first attempt of a `pipelineTask`) finished executing but failed. It has 2 retries pending. |
+| Unknown  | Running     | Second attempt of a `taskRun` has started to perform its work.                                             |
+| Unknown  | ToBeRetried | The `taskRun` (second attempt of a `pipelineTask`) finished executing but failed. It has 1 retry pending.  |
+| Unknown  | Running     | Third attempt of a `taskRun` has started to perform its work.                                              |
+| False    | Failed      | The `taskRun` (third attempt of a `pipelineTask`) finished executing but failed. No more retries pending.  |
+
+Pipeline controller can now rely on `ConditionSucceeded` set to `Failed` after all the retries are exhausted.
+
+`TaskRun` reconciler archives existing status into `retriesStatus` at the end of each Reconcile loop when the `TaskRun`'s `Failure` is detected,
+then the existing status of a `taskRun` such that a next attempt starts executing.
+
```

Comment:
@XinruZhang please add this statement back for clarity:

https://github.com/tektoncd/community/pull/882/files#diff-4a8da49e599db9e42576edb3fda3e041fd32e3f502ab42df54f8eff0d6e301b1R160

Reply:
Thanks Priti @pritidesai Added!